StormDBManager.py

```
@@ -0,0 +1,188 @@
+        result = connection.execute(query, arguments).get_one()
+        connection.close()
+        return result
+
+    @transact
+    def fetch_all(self, query, arguments=None):
+        """
+        Executes a query on the twisted thread-pool using the storm framework and
+        returns all results as a list of tuples.
+        :param query: The sql query to be executed.
+        :param arguments: Optional arguments that go with the sql query.
+        :return: A deferred that fires with a list of tuple results that matches the query or an empty list.
+        """
+        connection = Connection(self._database)
+        return connection.execute(query, arguments).get_all()
+
+    @transact
+    def insert(self, table_name, **argv):
+        """
+        Inserts data provided as keyword arguments into the table provided as an argument.
+        :param table_name: The name of the table the data has to be inserted into.
+        :param argv: A dictionary where the key represents the column and the value the value to be inserted.
+        :return: A deferred that fires when the data has been inserted.
+        """
+        connection = Connection(self._database)
+        self._insert(connection, table_name, **argv)
+        connection.close()
+
+    def _insert(self, connection, table_name, **argv):
+        """
+        Utility function to insert data which is not decorated by the @transact to prevent
+        a loop calling this function to create many threads.
+        Do NOT call this function on the main thread, it will be then blocking on that thread.
+        :param connection: The database connection object.
+        :param table_name: The name of the table the data has to be inserted into.
+        :param argv: A dictionary where the key represents the column and the value the value to be inserted.
+        :return: A deferred that fires when the data has been inserted.
+        """
+        if len(argv) == 0: return
```

Yeah it could :+1: Make the error explicit instead of swallowing it.